src/backends/gr.jl

```diff
@@ -1227,16 +1227,18 @@ function gr_display(sp::Subplot{GRBackend}, w, h, viewport_canvas)
             gr_text(GR.wctondc(xi, yi)..., str)
         end
 
-        # draw the colorbar
-        if cmap && st != :contour # special colorbar with steps is drawn for contours
-            gr_set_line(1, :solid, yaxis[:foreground_color_axis])
-            gr_set_transparency(1)
-            gr_colorbar(sp, clims)
-        end
-
         GR.restorestate()
     end
 
+    # draw the colorbar
+    GR.savestate()
```

Comment:
@jheinen is the `savestate` and `restorestate` necessary here?

Reply:
The `savestate()` ... `restorestate()` functions ensure, that the attributes are reset to the old settings after drawing the colorbar. They don't produce much overhead ...

Reply:
Great, thanks!